lib/media/streaming_engine.js

```diff
@@ -1953,6 +1959,29 @@ shaka.media.StreamingEngine = class {
     shaka.log.v2(logPrefix, 'appended media segment');
   }
 
+  /**
+   * @param {shaka.media.StreamingEngine.MediaState_} mediaState
+   * @param {shaka.media.SegmentReference|shaka.media.InitSegmentReference}
+   *   reference
+   * @return {!Promise}
+   * @private
+   */
+  async changeTypeIfNeeded_(mediaState, reference) {
+    // TODO: Do I also need to handle new init segments for this content, or
+    // do we handle that already? The test content I have is all
+    // self-initializing segments.
+    if (reference.codecs) {
+      // There's codecs information for this SegmentReference. If it's different
+      // from the current codecs of the buffer, changeType.
+      if (reference.codecs != mediaState.codecs) {
+        const fullMimeType = shaka.util.MimeUtils.getFullType(
+            mediaState.stream.mimeType, reference.codecs);
+        await this.playerInterface_.mediaSourceEngine.changeType(
```

Comment:
It might be better to call `mediaSourceEngine.setStreamProperties` and have it call `changeType`, instead of calling `changeType` directly. Not sure.

Reply:
Yes, it's better

Reply:
After looking over the code more, I think that `setStreamProperties` does too much to be called here. Instead, I decided to make `codecSwitchIfNecessary_` no longer private, and called that.

Reply:
Yeah, I moved the mimeType/codec changing into that call, and it looks a lot nicer.